scripts/github_api

```diff
@@ -0,0 +1,118 @@
+#!/bin/python2
+from functools import partial
+import argparse
+import json
+import re
+import requests
+import sys
+
+CLIENT_ERROR = 422
+API_BASE_URL = "https://api.github.com/"
+
+
+def request(*args, **kwargs):
+    raise Exception("The '_setup_auth' function must be called once "
+                    "before a request can be made")
+
+
+def _setup_auth(user, password):
+    auth = (user, password)
+    global request
+    request = partial(_request, auth)
+
+
+def _request(auth, method, endpoint, **kwargs):
```

Comment:
I have no idea what will be done with these keyword arguments. HTTP headers? Maybe method doc will make it clearer